moto/mediastore/models.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import unicode_literals
+from boto3 import Session
+from moto.core import BaseBackend, BaseModel
+from collections import OrderedDict
+from datetime import date
+from .exceptions import ResourceNotFoundException, PolicyNotFoundException
+
+
+class Container(BaseModel):
+    def __init__(self, *args, **kwargs):
+        self.arn = kwargs.get("arn")
+        self.name = kwargs.get("name")
+        self.endpoint = kwargs.get("endpoint")
+        self.status = kwargs.get("status")
+        self.creation_time = kwargs.get("creation_time")
+
+    def to_dict(self, exclude=None):
+        data = {
+            "ARN": self.arn,
+            "Name": self.name,
+            "Endpoint": self.endpoint,
+            "Status": self.status,
+            "CreationTime": self.creation_time,
+        }
+        if exclude:
+            for key in exclude:
+                del data[key]
+        return data
+
+
+class MediaStoreBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(MediaStoreBackend, self).__init__()
+        self.region_name = region_name
+        self._containers = OrderedDict()
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+        self._lifecycle_policies = OrderedDict()
+
+    def create_container(self, name, tags):
+        arn = "arn:aws:mediastore:container:{}".format(name)
+        container = Container(
+            arn=arn,
+            name=name,
+            endpoint="/{}".format(name),
+            status="CREATING",
+            creation_time=date.today().strftime("%m/%d/%Y, %H:%M:%S"),
+        )
+        self._containers[name] = container
+        return container
+
+    def describe_container(self, name):
+        if name not in self._containers:
+            raise ResourceNotFoundException()
+        container = self._containers[name]
+        container.status = "ACTIVE"
+        return container
+
+    def list_containers(self, next_token, max_results):
+        containers = list(self._containers.values())
+        response_containers = [c.to_dict() for c in containers]
+        return response_containers, next_token
+
+    def put_lifecycle_policy(self, container_name, lifecycle_policy):
+        if container_name not in self._containers:
+            raise ResourceNotFoundException()
+        self._containers[container_name].lifecycle_policy = lifecycle_policy
+        return {}
+
+    def get_lifecycle_policy(self, container_name):
+        try:
+            if container_name not in self._containers:
+                raise ResourceNotFoundException()
+            lifecycle_policy = self._containers[container_name].lifecycle_policy
+            return lifecycle_policy
+        except AttributeError:
```

Comment:
Instead of the try-except, I prefer the attribute to always exist, and do a None-check here:
```
class Container(BaseModel):
    def __init__(self, *args, **kwargs):
        self.lifecycle_policy = None

....
def get_lifecycle_policy(self, container_name):
    lifecycle_policy = self._containers[container_name].lifecycle_policy
    if not lifecycle_policy:
        raise PolicyNotFoundException()
```